extensions/vscode/src/api/types/contentRecords.ts

```diff
@@ -31,6 +31,13 @@ export type ContentRecordError = {
   state: ContentRecordState.ERROR;
 } & ContentRecordLocation;
 
+type OptionalPreDeploymentFields = {
+  id?: string;
+  bundleUrl?: string;
+  dashboardUrl?: string;
+  directUrl?: string;
+};
```

Comment:
I noticed that in `EvenEasierDeploy.vue` `isPreContentRecordWithID` and `isPreContentRecordWithoutID` was added.

We could potentially split up `PreContentRecord` into these two types to accomplish the same thing that is sort of being split up here (between type files and component files)

Reply:
Discussed with Jordan and we decided to hold off on this change. There are complications in implementing two additional types which then multiply into the other variants (such as with and without config).